tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -222,8 +256,8 @@ final class ScalaJSClassEmitter(semantics: Semantics, outputMode: OutputMode,
         case OutputMode.ECMAScript51Global | OutputMode.ECMAScript51Isolated =>
           genAddToPrototype(className, method.name, methodFun)
 
-        case OutputMode.ECMAScript6 =>
-          js.MethodDef(static = false, genPropertyName(method.name),
+        case OutputMode.ECMAScript6 | OutputMode.ECMAScript6StrongMode =>
+          js.MethodDef(static = method.static, genPropertyName(method.name),
               methodFun.args, methodFun.body)
```

Comment:
They're slower than top level functions, since they must be resolved as members of the class function, which could be tampered with by outside code.